src/EditorFeatures/CSharpTest/GenerateComparisonOperators/GenerateComparisonOperatorsTests.cs

```diff
@@ -0,0 +1,448 @@
+        public async Task TestMissingWithAllExistingOperators()
+        {
+            await TestMissingAsync(
+@"
+using System;
+
+class C : IComparable<C>
+{
+    public int CompareTo(C c) => 0;
+
+    public static bool operator <(C left, C right)
+    {
+        return left.CompareTo(right) < 0;
+    }
+
+    public static bool operator >(C left, C right)
+    {
+        return left.CompareTo(right) > 0;
+    }
+
+    public static bool operator <=(C left, C right)
+    {
+        return left.CompareTo(right) <= 0;
+    }
+
+    public static bool operator >=(C left, C right)
+    {
+        return left.CompareTo(right) >= 0;
+    }
+
+[||]
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateComparisonOperators)]
+        public async Task TestWithExistingOperator()
+        {
+            await TestInRegularAndScript1Async(
+@"
+using System;
+
+class C : IComparable<C>
+{
+    public int CompareTo(C c) => 0;
+
+    public static bool operator <(C left, C right)
+    {
+        return left.CompareTo(right) < 0;
+    }
+
+[||]
+}",
+@"
+using System;
+
+class C : IComparable<C>
+{
+    public int CompareTo(C c) => 0;
+
+    public static bool operator <(C left, C right)
+    {
+        return left.CompareTo(right) < 0;
+    }
+
+    public static bool operator >(C left, C right)
+    {
+        return left.CompareTo(right) > 0;
+    }
+
+    public static bool operator <=(C left, C right)
+    {
+        return left.CompareTo(right) <= 0;
+    }
+
+    public static bool operator >=(C left, C right)
+    {
+        return left.CompareTo(right) >= 0;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateComparisonOperators)]
+        public async Task TestMultipleInterfaces()
+        {
+            await TestInRegularAndScript1Async(
+@"
+using System;
+
+class C : IComparable<C>, IComparable<int>
+{
+    public int CompareTo(C c) => 0;
+    public int CompareTo(int c) => 0;
+
+[||]
+}",
+@"
+using System;
+
+class C : IComparable<C>, IComparable<int>
+{
+    public int CompareTo(C c) => 0;
+    public int CompareTo(int c) => 0;
+
+    public static bool operator <(C left, int right)
+    {
+        return left.CompareTo(right) < 0;
+    }
+
+    public static bool operator >(C left, int right)
+    {
+        return left.CompareTo(right) > 0;
+    }
+
+    public static bool operator <=(C left, int right)
+    {
+        return left.CompareTo(right) <= 0;
+    }
+
+    public static bool operator >=(C left, int right)
+    {
+        return left.CompareTo(right) >= 0;
+    }
+}", index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateComparisonOperators)]
+        public async Task TestInInterfaceWithDefaultImpl()
+        {
+            await TestInRegularAndScript1Async(
+@"
+using System;
+
+interface C : IComparable<C>
```

Comment:
does this also work if we do ?

```csharp
interface C<T> : IComparable<T>
{
    int IComparable<T>.CompareTo(C<T> c) => 0;
}

class D : C<D>
{
   [||]
}
```

Reply:
it does not.  primarily because i only do this on the declared interfaces of the type itself (not inherited interfaces).  I figured it was unlikely you'd want the derived to have them be distinct.   If we hear feedback about this we could change (but i find it unlikely).